src/linear_algebra/affine_space/independent.lean

```diff
@@ -471,6 +472,55 @@ lemma affine_independent_iff {ι} {p : ι → V} :
   ∀ (s : finset ι) (w : ι → k), s.sum w = 0 → ∑ e in s, w e • p e = 0 → ∀ (e ∈ s), w e = 0 :=
 forall₃_congr (λ s w hw, by simp [s.weighted_vsub_eq_linear_combination hw])
 
+/-- Given an affinely independent family of points, a weighted subtraction lies in the
+`vector_span` of two points given as affine combinations if and only if it is a weighted
+subtraction with weights a multiple of the difference between the weights of the two points. -/
+lemma weighted_vsub_mem_vector_span_pair {p : ι → P} (h : affine_independent k p)
+  {w w₁ w₂ : ι → k} {s : finset ι} (hw : ∑ i in s, w i = 0) (hw₁ : ∑ i in s, w₁ i = 1)
+  (hw₂ : ∑ i in s, w₂ i = 1) :
+  s.weighted_vsub p w ∈
+    vector_span k ({s.affine_combination p w₁, s.affine_combination p w₂} : set P) ↔
+    ∃ r : k, ∀ i ∈ s, w i = r * (w₁ i - w₂ i) :=
+begin
+  rw mem_vector_span_pair,
+  refine ⟨λ h, _, λ h, _⟩,
+  { rcases h with ⟨r, hr⟩,
+    refine ⟨r, λ i hi, _⟩,
+    rw [s.affine_combination_vsub, ←s.weighted_vsub_const_smul, ←sub_eq_zero, ←map_sub] at hr,
+    have hw' : ∑ j in s, (r • (w₁ - w₂) - w) j = 0,
+    { simp_rw [pi.sub_apply, pi.smul_apply, pi.sub_apply, smul_sub, finset.sum_sub_distrib,
+               ←finset.smul_sum, hw, hw₁, hw₂, sub_self] },
+    have hr' := h s _ hw' hr i hi,
+    rw [eq_comm, ←sub_eq_zero, ←smul_eq_mul],
+    exact hr' },
+  { rcases h with ⟨r, hr⟩,
+    refine ⟨r, _⟩,
+    let w' := λ i, r * (w₁ i - w₂ i),
+    change ∀ i ∈ s, w i = w' i at hr,
+    rw [s.weighted_vsub_congr hr (λ _ _, rfl), s.affine_combination_vsub,
+        ←s.weighted_vsub_const_smul],
+    congr }
+end
+
+/-- Given an affinely independent family of points, an affine combination lies in the
+span of two points given as affine combinations if and only if it is an affine combination
+with weights those of one point plus a multiple of the difference between the weights of the
+two points. -/
+lemma affine_combination_mem_affine_span_pair {p : ι → P} (h : affine_independent k p)
+  {w w₁ w₂ : ι → k} {s : finset ι} (hw : ∑ i in s, w i = 1) (hw₁ : ∑ i in s, w₁ i = 1)
+  (hw₂ : ∑ i in s, w₂ i = 1) :
+  s.affine_combination p w ∈
+    line[k, s.affine_combination p w₁, s.affine_combination p w₂] ↔
+    ∃ r : k, ∀ i ∈ s, w i = r * (w₂ i - w₁ i) + w₁ i :=
```

Comment:
This is really the result that an affine combination of affine combinations is an affine combination, for the case when the family has two elements.

Or to put it another way, I suppose this is the binary version of a result whose pi version would be something like:
```
import linear_algebra.affine_space.combination

variables {ι ι' k V P : Type*} [ring k] [add_comm_group V] [module k V] [S : add_torsor V P]
  (p : ι' → P) (w : ι → k) (w' : ι → ι' → k) (s : finset ι) (s' : finset ι')

-- This will be true if the various weights sum to one
#check s.affine_combination (λ j, s'.affine_combination p (w' j)) w =
  s'.affine_combination p (λ j, s.affine_combination (λ i, w' i j) w)
```

Reply:
It might be nice to add this at some point.

Reply:
Actually that's the pi version of a slightly different result (all the versions are appropriate for mathlib). The binary (`line_map`) or pi affine combination of affine combinations is an affine combination with the obvious weights, regardless of whether the points are affinely independent, and any point in the affine span can be expressed as a combination with weights in that form. The result I have here, that requires affine independence, says that an affine combination is in the affine span if and only if the weights are of that form; without affine independence, the same point could also be expressed as an affine combination with weights not in that form.